flocker/node/test/test_change.py

```diff
@@ -521,3 +523,47 @@ def test_failed(self, logger):
         action._logger = logger
         failure = self.failureResultOf(run_state_change(action, DEPLOYER))
         self.assertEqual(failure.getErrorMessage(), "Oh no")
+
+    def assert_nested_logging(self, combo, action_type, logger):
+        """
+        All the underlying ``IStateChange`` will be run in Eliot context in
+        which the sequential ``IStateChange`` is run, even if they are not
+        run immediately.
+
+        :param combo: ``sequentially`` or ``in_parallel``.
+        :param action_type: ``eliot.ActionType`` we expect to be parent of
+            sub-changes logs.
+        :param logger: A ``MemoryLogger`` where messages go.
+        """
+        actions = [ControllableAction(result=Deferred()),
+                   ControllableAction(result=succeed(None))]
+        for action in actions:
+            self.patch(action, "_logger", logger)
```

Comment:
> `patch` is better insofar it complains if attribute isn't there in first place.

Is that particularly relevant here? We assign to instance variables elsewhere without `patch`. 

> I could fix that in this branch. Or maybe we want follow up branch where we cleanup that and the property thing.

Well, my preference is that you fix them here, but I'm also happy for them to be fixed in a follow-up branch.